### PR TITLE
💡 Add Lightbox

### LINKED ISSUE #120: Make the gallery for entry photos
## Learning objectives

- Objective

## Todo

These points are a rough guideline. **Please** feel free to discuss with others on the team about the best way to design this component!

- Todo

## Useful links

- Useful link

### PR DESCRIPTION
**Pull request by code-arman**
## Status:
:rocket: Ready

## Description
Allow users to view review images in full screen (see figma for design and entry 3383 for demo)

Fixes #120 

## Todos
- Fix eslint issue
- Fix bug when going back
- Different mobile design?

## Screenshots

https://user-images.githubusercontent.com/46976119/142603447-058c8912-7eb3-40f9-b799-490aae5879c9.mov

### CODE REVIEW

File: src/utils/api.ts [hunk 1]
```diff
@@ -91,9 +91,133 @@ export const getTypeById = (
   id: paths['/types/{id}']['get']['parameters']['path']['id'],
 ) => handleResponse(instance.get(`/types/${id}`))
 
-export const getReviews = (
```
Comment (SirajChokshi) [inline src/utils/api.ts:94]: Great data mock for demo-ing, but lets remove any changes made to this file for prod

File: src/components/entry/Lightbox.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+//  has image on top and reviews on bottom?
+const StyledReviewImage = styled.img`
+  background-color: black;
+  object-fit: contain;
+  object-position: center;
+  width: calc(35vw - 32px);
+  min-width: 300px;
+  height: 500px;
+`
+const ReviewContainer = styled.div`
```
Comment (SirajChokshi) [inline src/components/entry/Lightbox.js:24]: We're missing a border-radius on this element.

File: src/components/entry/Lightbox.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+  onDismiss,
+  review,
+  currReviewIndex,
+  reviewSubImgIndex,
+  setCurrReviewIndex,
+  setReviewSubImgIndex,
+  reviewImages,
+}) => {
+  const incrementReviewImage = () => {
+    reviewSubImgIndex + 1 < reviewImages[currReviewIndex].length
```
Comment (SirajChokshi) [inline src/components/entry/Lightbox.js:85]: this function is a little too pithy, maybe we could use some if/else so its a little more clear what's going on

File: src/components/entry/Lightbox.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+import Review, { StyledImagePreview } from './Review'
+
+const StyledDialog = styled(Dialog)`
+  display: flex;
+  width: 80%;
+  max-width: fit-content;
+`
+//  TODO: Add media queries for mobile so that the layout of
+//  has image on top and reviews on bottom?
+const StyledReviewImage = styled.img`
```
Comment (SirajChokshi) [inline src/components/entry/Lightbox.js:16]: We're missing a border radius on the image preview

File: src/components/entry/Lightbox.js [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+import styled from 'styled-components/macro'
+
+import ImagePreview from '../ui/ImagePreview'
+import ResetButton from '../ui/ResetButton'
+import Review, { StyledImagePreview } from './Review'
+
+const StyledDialog = styled(Dialog)`
+  display: flex;
+  width: 80%;
+  max-width: fit-content;
```
Comment (SirajChokshi) [inline src/components/entry/Lightbox.js:12]: The implication of this max-width is that the children determine the size of the parent. This seems to work, but lets think more about what that suggests about how the Dialog should work (should it fit content or fit the page as a whole?)

File: src/components/entry/Lightbox.js [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+        <ExitButton onClick={onDismiss}>
+          <X size={30} />
+        </ExitButton>
+        <Review review={review[currReviewIndex]} includePreview={false} />
+
+        <ThumbnailImages>
+          {reviewImages[currReviewIndex].map((photo) =>
+            photo.thumb ===
+            reviewImages[currReviewIndex][reviewSubImgIndex].thumb ? (
+              <SelectedImagePreview $small key={photo.thumb}>
```
Comment (SirajChokshi) [inline src/components/entry/Lightbox.js:135]: Maybe we could use a wrapped version of `SelectedImagePreview` for both selected and unselected instances and have a prop that dictates if an instance is currently selected.

**Review by SirajChokshi — CHANGES_REQUESTED** [review]
Great work implementing this so far -- its definitely almost there.

A few things I noticed:
* Missing border-radii on image and modal
* Margin is too large on modal
Screen Shot 2021-11-19 at 4 06 40 AM

- In the above screenshot, the border looks off (reference the designs)
- The thumbnails beneath the review should be clickable

Left a few comments about code-style.

**Review by Qrtn — APPROVED** [review]

**Review by daniel-moon32 — APPROVED** [review]
Looks good!

**Pull request merged**